### PR TITLE
feat: connector support http export transport

### PR DESCRIPTION
**Pull request by luongngocminh**
### Description

Provide HTTP Transport export type for connector.
Also provide a way for user to configure how the output data will be: JSON or Protobuf(default)
Protobuf will be delivered with the Content-Type of "application/octet-stream"

### Related Issue
#223 

### Checklist

- [x] I have tested the changes locally.
- [x] I have reviewed the code changes.
- [x] I have updated the documentation, if necessary.
- [x] I have added appropriate tests, if applicable.

### Screenshots
image

### Additional Notes
The JSON output format is still not very friendly for parsing. Adjacent tagging is preferred. But due to the limitation of the `prost` and `prost-build` crate, serde tagging is difficult to achieve.

## Summary by CodeRabbit

- **New Features**
	- Enhanced serialization capabilities with support for `serde::Serialize`.
	- Added `Format` support in the media server, enabling data transmission over `http` and `https` using `JSON` and `Protobuf` formats.
	- Introduced `HttpTransporter` for asynchronous data transmission over HTTP.
- **Enhancements**
	- Extended transporter selection logic to support `http` and `https` protocols.
	- Improved `NatsTransporter` with serialization format flexibility.
- **Documentation**
	- Updated comments to clarify `http://host:port` as a valid format for `mq_uri`.

### CODE REVIEW

File: servers/media-server/src/server/connector.rs [hunk 1]
```diff
@@ -74,11 +77,20 @@ where
...
+            let nats = NatsTransporter::<MediaEndpointLogRequest>::new(&opts.mq_uri, &opts.mq_channel, &opts.format)
                 .await
                 .expect("Nats should be connected");
             Box::new(nats)
         }
+        "http" => {
+            let http = transports::http::HttpTransporter::<MediaEndpointLogRequest>::new(&opts.mq_uri, &opts.format);
+            Box::new(http)
+        }
+        "https" => {
```
Comment (giangndm) [inline servers/media-server/src/server/connector.rs:89]: should merge https and https like:

```rust
"http" | "https" => {
         //logic here
}
```

File: servers/media-server/src/server/connector.rs [hunk 2]
```diff
@@ -33,7 +32,7 @@ use super::MediaServerContext;
 #[derive(Parser, Debug)]
 #[command(author, version, about, long_about = None)]
 pub struct ConnectorArgs {
-    /// Message Queue URI in the form of `amqp://user:pass@host:port/vhost`
+    /// Message Queue URI in the form of `http://host:port` or `nats://host:port`
```
Comment (giangndm) [inline servers/media-server/src/server/connector.rs:35]: maybe we should setting http as default, nats is only for complex system. I guest that most of user will use http instead

File: servers/media-server/src/server/connector/transports/mod.rs [hunk 3]
```diff
@@ -1,17 +1,34 @@
...
     InvalidURI,
 }
 
+#[derive(Debug, ValueEnum, Clone)]
+pub enum Format {
+    Json,
+    Protobuf,
+}
+
+impl Display for Format {
```
Comment (giangndm) [inline servers/media-server/src/server/connector/transports/mod.rs:21]: maybe we should switch to [derive_more](https://crates.io/crates/derive_more) for simpler

File: servers/media-server/src/server/connector/transports/http.rs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+#[async_trait]
+impl<M: Message + Serialize> ConnectorTransporter<M> for HttpTransporter<M> {
+    async fn send(&mut self, data: M) -> Result<(), io::Error> {
+        match self.format {
+            Format::Json => {
+                self.client.post(&self.url).json(&data).send().await.map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
+            }
+            Format::Protobuf => {
+                self.client
```
Comment (giangndm) [inline servers/media-server/src/server/connector/transports/http.rs:35]: we should check response status is success or not

**Review by giangndm — CHANGES_REQUESTED** [review]

File: servers/media-server/src/server/connector.rs (hunk 2 above)
Comment (luongngocminh) [reply servers/media-server/src/server/connector.rs:35]: but what will be the default value, just a arbitrary localhost URL will a random common port?

**Review by giangndm — APPROVED** [review]
Look good

**Pull request merged**